File: Tutorial5-3bDagger-Android/src/main/java/com/example/tutorial5_1daggerandroid/di/ToastMakerModule.java
package com.example.tutorial5_1daggerandroid.di;


import android.content.Context;

import com.example.tutorial5_1daggerandroid.ToastMaker;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

// This is SubComponent Module

@Module
public class ToastMakerModule {

    @ActivityScope
    @Provides
    ToastMaker provideToastMaker(Context context) {
        System.out.println("ToastMakerModule context: " + context);
        return new ToastMaker(context);
    }
}

File: Tutorial4-6SubComponentBuilders-MultipleModulesAndActvities/src/main/java/com/example/tutorial4_6subcomponentbuilders_multiplemodulesandactvities/di/ToastMakerModule.kt
package com.example.tutorial4_6subcomponentbuilders_multiplemodulesandactvities.di

import android.content.Context
import com.example.tutorial4_6subcomponentbuilders_multiplemodulesandactvities.di.scope.FragmentScope
import com.example.tutorial4_6subcomponentbuilders_multiplemodulesandactvities.model.ToastMaker
import dagger.Module
import dagger.Provides

/**
 * This is SubComponent Module of [ToastAndPreferencesSubComponent] in this example
 * A module can be used in multiple components
 *
 * Sub component and module CAN NOT have different scopes.
 * IMPORTANT: Both can be un- scoped but cannot have with different scopes
 */
@Module
class ToastMakerModule {

    @FragmentScope
    @Provides
    fun provideToastMaker(context: Context): ToastMaker {
        return ToastMaker(context)
    }

}